app/models/chapter.rb

```diff
@@ -80,11 +79,9 @@ def fix_positions
   after_save :invalidate_chapter_count,
     if: Proc.new { |chapter| chapter.saved_change_to_posted? }
 
-  after_save :expire_cache_on_coauthor_removal
-
   before_destroy :fix_positions_after_destroy, :invalidate_chapter_count
   def fix_positions_after_destroy
-    if work && position
+    if work&.persisted? && position
```

Comment:
Is this because work now has `has_many :chapters, inverse_of: :work, autosave: true`?

Reply:
Sort of. I added it because at the time I was worried that the inverses would break deleting works, but after some poking, I'm not sure it's necessary. According to the Rails docs, the inverse name can be inferred if it matches the name of the class that it's defined in, so the inverse is already working, and `inverse_of: :work` is pretty much just a formality. The only benefit of checking `persisted?` at this point is that it decreases the number of SQL queries required during work deletion.

I can revert the change, if you'd prefer.

Reply:
Fewer SQL queries sound good, let's keep it.